sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -365,16 +365,17 @@ case class Join(
 }
 
 /**
- * Append data to an existing table.
+ * Base trait for DataSourceV2 write commands
  */
-case class AppendData(
-    table: NamedRelation,
-    query: LogicalPlan,
-    isByName: Boolean) extends LogicalPlan {
+trait V2WriteCommand extends Command {
+  def table: NamedRelation
+  def query: LogicalPlan
+
   override def children: Seq[LogicalPlan] = Seq(query)
-  override def output: Seq[Attribute] = Seq.empty
 
-  override lazy val resolved: Boolean = {
+  override lazy val resolved: Boolean = writeResolved
+
+  def writeResolved: Boolean = {
```

Comment:
`inputPlansResolved`?

Reply:
Why?

When you ask for changes, please include your rationale so that I understand the motivation and we can choose an appropriate solution.

Reply:
What does `writeResolved` mean? The plans for writing are resolved?

Reply:
It means that the write is resolved: the incoming query matches the output relation. I don't think that `inputPlansResolved` is clear because the output relation is not an input plan.

I'll rename this to `outputResolved`, which I think is more clear.